payments/views.py

```diff
@@ -0,0 +1,107 @@
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from courses.models import Curriculum
+
+from .models import Cart, CartItem
+from .serializers import CartSerializer, CartItemSerializer
+
+
+class CartItemView(APIView):
+
+    def get_cart(self, user):
+        """
+        사용자별 장바구니를 조회합니다. 없으면 새로 생성합니다.
+        """
+        cart, _ = Cart.objects.get_or_create(user=user)
+        return cart
+
+    def get_cart_item(self, cart, item_id):
+        """
+        사용자별 장바구니에 있는 특정 상품을 조회합니다.
+        """
+        try:
+            return CartItem.objects.get(id=item_id, cart=cart)
+        except CartItem.DoesNotExist:
+            return Response(
+                {"detail": "카트에 상품이 없습니다."}, status=status.HTTP_404_NOT_FOUND
+            )
+
+    def get(self, request, item_id=None):
+        """
+        장바구니에 있는 상품들을 조회하며, item_id가 주어지면 개별 상품을 조회합니다.
+        """
+        cart = self.get_cart(request.user)
+
+        if item_id is None:
+            serializer = CartSerializer(cart)
+        else:
+            cart_item = self.get_cart_item(cart, item_id)
+            if cart_item is None:
+                return Response(
+                    {"detail": "장바구니에 상품이 없습니다."},
+                    status=status.HTTP_404_NOT_FOUND,
+                )
+            serializer = CartItemSerializer(cart_item)
+
+        return Response(serializer.data)
+
+    def post(self, request):
+        """
+        장바구니에 새 상품을 추가합니다.
+        """
+        cart = self.get_cart(request.user)
+        curriculum_id = request.data.get("curriculum_id")
+        quantity = int(request.data.get("quantity", 1))
+
+        if quantity < 0:
+            return Response(
+                {"detail": "수량은 0 이상이어야 합니다."},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        try:
+            curriculum = Curriculum.objects.get(id=curriculum_id)
+        except Curriculum.DoesNotExist:
+            return Response(
+                {"detail": "해당되는 커리큘럼이 없습니다."},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        cart_item, created = CartItem.objects.update_or_create(
+            cart=cart,
+            curriculum=curriculum,
+            defaults={"quantity": quantity},
+        )
+
+        serializer = CartItemSerializer(cart_item)
+        response_data = serializer.data
+        if created:
+            response_data["message"] = "장바구니에 새로운 상품이 추가되었습니다."
+        else:
+            response_data["message"] = "장바구니의 상품이 업데이트되었습니다."
+
+        return Response(
+            response_data,
+            status=status.HTTP_201_CREATED if created else status.HTTP_200_OK,
+        )
```

Comment:
이렇게 하면 if를 한번 보면 되니까 이런 방법이 가독성이 조금 좋다고 생각합니다.

```suggestion
        if created:
            return ...
        else:
            return ...
```

Reply:
if 한번만에 내용을 다 보일 수 있긴 한데, 그러면 status를 두번 작성하게 되면서 코드가 좀 더 길어지지 않을까요?